src/cards/wakatime-card.js

```diff
@@ -304,17 +308,32 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
             noCodingActivityNode({
               // @ts-ignore
               color: textColor,
-              text: i18n.t("wakatimecard.nocodingactivity"),
+              text: !stats.is_coding_activity_visible
+                ? i18n.t("wakatimecard.notpublic")
+                : stats.is_other_usage_visible
+                ? i18n.t("wakatimecard.nocodingactivity")
+                : i18n.t("wakatimecard.nocodedetails"),
             }),
           ],
       gap: lheight,
       direction: "column",
     }).join("");
   }
 
+  // Get title range text
+  let titleText = i18n.t("wakatimecard.title");
+  switch (stats.range) {
```

Comment:
It seems that the `range` parameter has been removed, so will `stats.range` always be undefined here? Since we did not actually assign it during the loop, and it might not be provided by the user either. If so, maybe we also need to modify the relevant mock data (and their expected responses) in the e2e test, as one of them seems to have failed (and `range` parameter should not be provided in mock data).

Reply:
`range` attribute is still returned in `Stats` endpoint:

https://wakatime.com/developers/#stats

Reply:
> `range` attribute is still returned in `Stats` endpoint:
> 
> https://wakatime.com/developers/#stats

Get it. When debugging the failure in e2e test, I thought `range` was from the query by mistake. Thanks for pointing out!

Reply:
It seems like [catelinemnemosyne](https://wakatime.com/@catelinemnemosyne) is using the `range` option `all_time` now, which causes the e2e test to fail, so simply modify the mock data should be fine.

Reply:
Yea, GitHub totally lost it for some minutes 🤣. I think I fixed all your issues in [98fcc65](https://github.com/anuraghazra/github-readme-stats/pull/2707/commits/98fcc651138c0d6ae5cc68c3dbf8cc58849ccf75).